### DISCUSSION TITLE
how do i use this thing

### DISCUSSION THREAD

**Discussion by pleasegivemeafry** [pspdev/pspdev#102] [Q&A] [answered]
This is my first post(?) on github so sorry if it sounds a bit unprofessional. but i followed [this](https://pspdev.github.io/installation/windows.html) tutorial and i did everything, and when i type:

> psp-config --pspdev-path

it says:

> home/myusernameimnotgonnasayit/pspdev

so i assume it worked. the problem here is that i just don't know how to use the pspdev thingy. i wanted to test out [one of these codes](https://pspdev.github.io/basic_programs.html) so i could see if it actually works and can be indeed used for psp homebrew development, but i just don't know how to _put_ the code in the thing and build yk? i've never had experience with this kind of thing in the past so help might be appreciated. thanks!

P.S. i don't know how to write code in c, i am currently trying to learn lua so i can have a bit of coding experience before jumping onto c, idk i feel like i need that before learning it.
another P.S. i don't have an actual psp i just have the emulator (yk which one) i'm just saying this incase there's something i need to do before testing out the homebrews so yeah any help in advance is cool ^__^

**Comment by sharkwouter** [comment]
Hey @pleasegivemeafry, welcome to the community. You managed to get the PSPDEV toolchain up and running and building code, that's good. You can use it to make C and C++ based applications. If you're working on a Lua program you won't be needing it.

If you want access to the files in the Ubuntu command line, there is a small paragraph on the install page which tells you how to do that, it says the following:
```
Files in Ubuntu can be accessed through a network share.
In This Computer right click on the background and select Map network drive.
Set the folder to \\wsl$ to make it easily accessible.
``` 

After you've done this, you can open the directory that was created and go to home and then your user name. That is where you are when you open a terminal. You can open files from there in a text editor like VSCode.

I think it would be good to talk a bit about using a code editor on the website, since people might be new.

**Reply by pleasegivemeafry** [reply]
thanks for your answer! but what i was specifically asking was how to build the "main.c" and "CMakeLists.txt" files from the basic programs page. i'm assuming they're both separate files that have to be made outside of the ubuntu shell, but i don't know how to build them inside of the shell.

_..tbh i should have just asked how to build the basic programs LOOOL_

**Reply by sharkwouter** [reply, marked as answer]
You can just create a new directory and create the files using your favorite editor. You can navigate to there by using the cd command in the Ubuntu terminal. If you run `cd /mnt/c/Users/` you go to the users directory on your C drive for example.

**Reply by sharkwouter** [reply]
Of you can hold shift and right click on the directory in explorer and select `Open Linux shell here.`. That would also work. Then you can run the build commands.

**Reply by pleasegivemeafry** [reply]
it works!! i made a folder on the pspdev directory and pasted both files there, then i ran the build command and it worked!!!! thank you so much ^^

**Reply by sharkwouter** [reply]
Great, you're welcome!
